Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -176,27 +189,64 @@ def generate_default_port_by_dialect(dialect: str) -> Optional[str]:
     return None
 
 
-def generate_bind_vars(bind_variables_names: str, bind_variables_values: str) -> Any:
+def generate_variable_names_and_mapping(bind_variables_values_list: list, query: str, dialect: str) ->\
+        tuple[dict[str, Any], str | Any]:
+    """
+    In case of passing just bind_variables_values, since it's no longer supported in SQL Alchemy v2.,
+    this function generates names for those variables and return an edited query with a mapping.
+    Args:
+        bind_variables_values_list: Values to put in the bind variables
+        query: The given query which contains chars to replace
+        dialect: The DB dialect
+
+    Returns: A mapping (dict) and an edited query.
+
+    """
+    # For counting and replacing, re.findall needs "//?", whereas replace needs "?"
+    mapping_dialect_regex = {MICROSOFT_SQL_SERVER: ("\\?", "?"),
+                             MS_ODBC_DRIVER: ("\\?", "?"),
+                             POSTGRES_SQL: ("%s", "%s"),
+                             MY_SQL: ("%s", "%s"),
+                             ORACLE: ("%s", "%s")
+                             }
+    try:
+        char_to_count, char_to_replace = mapping_dialect_regex[dialect]
+    except Exception as e:
+        raise DemistoException(
+            "Placeholder was not found for the given dialect"
```

Comment:
please add a unit test with an unsupported dialect, make sure an exception is raised.

Reply:
Because dialect is a configuration parameter with multiple choices, there is no option to access a dialect we don't support, so I removed the exception and the condition.
No need for UT with case of exception